src/pyk/kcfg/kcfg.py

```diff
@@ -917,3 +929,16 @@ def reachable_nodes(self, source_id: str, *, reverse: bool = False, traverse_cov
                 worklist.extend(succ.source for succ in self.predecessors(target_id=node.id, covers=traverse_covers))
 
         return visited
+
+
+def path_length(_path: Iterable[KCFG.Successor]) -> int:
+    _path = list(_path)
```

Comment:
A small optimization you can do here:

```suggestion
    _path = tuple(_path)
```

then if `_path` is a `tuple`, then no copying will be performed.

Reply:
Done.